ibm/resource_ibm_firewall_shared.go

```diff
@@ -0,0 +1,240 @@
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/helpers/product"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/sl"
+)
+
+const (
+	FwHardwarePackageType = "ADDITIONAL_SERVICES_FIREWALL"
+)
+
+func resourceIBMFirewallShared() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMFirewallSharedCreate,
+		Read:     resourceIBMFirewallSharedRead,
+		Delete:   resourceIBMFirewallSharedDelete,
+		Exists:   resourceIBMFirewallSharedExists,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"billing_item_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"firewall_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAllowedStringValue([]string{"10MBPS_HARDWARE_FIREWALL", "20MBPS_HARDWARE_FIREWALL", "100MBPS_HARDWARE_FIREWALL", "1000MBPS_HARDWARE_FIREWALL"}),
+			},
+			"guest_type": {
```

Comment:
since you used virtual_instance_id & hardware_instance_id then we don't need guest type parmeter...
you can use if virtualId > 0 and else in createMethod